addons/l10n_it_edi/models/account_invoice.py

```diff
@@ -136,16 +136,14 @@ def _l10n_it_edi_prepare_fatturapa_tax_details(self, tax_details, reverse_charge
             tax_rate = tax_dict['tax'].amount
             expected_base_amount_currency = tax_amount_currency * 100 / tax_rate if tax_rate else False
             expected_base_amount = tax_amount * 100 / tax_rate if tax_rate else False
-            # Constraints within the edi make local rounding on price included taxes a problem.
+            # Constraints within the edi make local rounding on taxes a problem. There is a 0.01 € tolerance.
             # To solve this there is a <Arrotondamento> or 'rounding' field, such that:
             #   taxable base = sum(taxable base for each unit) + Arrotondamento
-            if tax_dict['tax'].price_include and tax_dict['tax'].amount_type == 'percent':
-                if expected_base_amount_currency and float_compare(base_amount_currency, expected_base_amount_currency, 2):
-                    tax_dict['rounding'] = base_amount_currency - (tax_amount_currency * 100 / tax_rate)
-                    tax_dict['base_amount_currency'] = base_amount_currency - tax_dict['rounding']
-                if expected_base_amount and float_compare(base_amount, expected_base_amount, 2):
-                    tax_dict['rounding_euros'] = base_amount - (tax_amount * 100 / tax_rate)
-                    tax_dict['base_amount'] = base_amount - tax_dict['rounding_euros']
+            if tax_dict['tax'].amount_type == 'percent' and tax_rate and abs(base_amount_currency - expected_base_amount_currency) > 0.01:
```

Comment:
this looks good to me, however I do have one concern..
the removal of `if expected_base_amount and float_compare(base_amount, expected_base_amount, 2):` might be an issue. 

My reasoning is that the difference could be greater than the tolerance in the invoice currency, but not in the company currency (or vice versa).

So in one particular scenario, where the invoice is a non-domestic reverse charge invoice, when the currency of the invoice is in e.g. CHF, and the company currency is EUR (as it always should be).
Imagine the difference between the base amount and the expected base amount in CHF does not exceed the tolerance, but it does exceed the tolerance when converted to EUR, we could end up with a situation in which the rounding is not performed.
The problem is that the content of the xml is always going to be converted to EUR, in which case the rounding has not been performed, and we have problems.

hopefully that was clear, I know it's a bit of a niche case, but considering the bug from which this issue arose was related to a non-domestic (though admittedly EUR) reverse-charge invoice, I can forsee it happening, and confusing us in the future

sorry for the essay

Reply:
@jco-odoo thoughts?

Reply:
You're right, i inverted currency and non currency. The doc specify that the tolerance is 0.01 €.

Reply:
ah you're right.. I didn't think that all the way though